dockerfiles/SynapseWorkers.Dockerfile

```diff
@@ -0,0 +1,64 @@
+# This dockerfile builds on top of Dockerfile-worker and includes a built-in postgres instance
+# as well as sets up the homeserver so that it is ready for testing via Complement
+FROM matrixdotorg/synapse:workers
+
+# Tell Complement that we are using its custom CA
+ENV COMPLEMENT_CA=true
+
+# Install postgresql
+RUN apt-get update
+RUN apt-get install -y postgresql
+
+# Configure a user and create a database for Synapse
+RUN pg_ctlcluster 11 main start &&  su postgres -c "echo \
+ \"ALTER USER postgres PASSWORD 'somesecret'; \
+ CREATE DATABASE synapse \
+  ENCODING 'UTF8' \
+  LC_COLLATE='C' \
+  LC_CTYPE='C' \
+  template=template0;\" | psql" && pg_ctlcluster 11 main stop
+
+# Modify the shared homeserver config with postgres support, certificate setup
+# and the disabling of rate-limiting
+COPY synapse/workers-shared.yaml /conf/workers/shared.yaml
+
+# Set up TLS certificates using the custom CA
+COPY keys/* /ca/
```

Comment:
Not following this here. Complement will auto-mount a volume to `/ca` which will contain the certs. Why are we doing this copy from a `/keys` directory?

Reply:
Good point. This was just copied over from the existing Synapse config, and can go. However I did notice that having `ENV COMPLEMENT_CA=true` in the dockerfile (at least in this setup) doesn't seem to goad Complement into running in CA mode.

Including the variable when running the dockerfile works as expected.

Reply:
`COMPLEMENT_CA=true` is a config option for running Complement itself, it's driven by the end-user not by random dockerfiles. It's up to the dockerfile to look in `/ca` if it exists to know whether to use it or not.

Reply:
I see, I was confused by the "or the docker container" bit in the README then: https://github.com/matrix-org/complement#complement-pki.